scrabble.js

```diff
@@ -1,8 +1,110 @@
 var Scrabble = function() {};
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+Scrabble.prototype.score = function(word) {
+
+  var scoring_chart = {
+    "A": 1,
+    "B": 3,
+    "C": 3,
+    "D": 2,
+    "E": 1,
+    "F": 4,
+    "G": 2,
+    "H": 4,
+    "I": 1,
+    "J": 8,
+    "K": 5,
+    "L": 1,
+    "M": 3,
+    "N": 1,
+    "O": 1,
+    "P": 3,
+    "Q": 10,
+    "R": 1,
+    "S": 1,
+    "T": 1,
+    "U": 1,
+    "V": 4,
+    "W": 4,
+    "X": 8,
+    "Y": 4,
+    "Z": 10
+  };
+
+  // converts input to a string and makes the input case-insensitive
+  word = String(word).toUpperCase();
+
+  //catches the first instance of anything that's NOT a letter
+  if (/[^a-z]/i.test(word) === true) {
+    throw "Word should only contain letters of the English alphabet.";
```

Comment:
The ^ in the regex group means "not". So, anything that's not a-z. I think it's the same as doing \W in a regex... 
It's just to handle those situations where people try to pass something that isn't an alphabet character.